spec/quotas/quotas.mdown

```diff
@@ -32,51 +35,95 @@ This document defines one additional capability URI.
+It explains the entities this value applies to. Some custom specifications might provide some additional values. If the
+client does not specify custom scope specifications in the "using" parameter of the request, the server should respond
+the JSON value `null`, instead of answering a scope value that the client does not support. Standard values are:
+
 * `account`: Applies for this account
 * `domain`: All users of this domain share this part of the quota
 * `global`: All users of this mail server share this part of the quota
 
+## The ResourceType Data Type
+
+The **ResourceType** is a `String` from an enumeration defined list of values, handled by the server.
+
+A resource type is like an unit of measure for the quota usage. Some custom specifications might provide some additional
+values. If the client does not specify custom resource type specifications in the "using" parameter of the request,
+the server should respond the JSON value `null`, instead of answering a resource type value that the client does not
+support. Standard values are:
+
+* `count`: The quota is measured in number of data type objects. For example, a quota can have a limit of 50 `Mail` objects.
+* `size`: The quota is measured in size. The default unit is in `bytes`, but a server can decide of the unit it wants
+to use (like in `octets`). For example, a quota can have a limit of 25000 `bytes`
+
 ## The Quota Object
 
 The quota object MUST contain the following fields:
 
 * **id**: `Id` The unique identifier for this object. It should respect the JMAP ID datatype defined in section 1.2 of [@!RFC8620]
 
-* **type**: `String` The type of quota. It should either be "messageCount" or "messageStorageSize". There is two ways we can define a quota: either on a number of emails, or in the amount of bytes.
+* **resourceType**: `ResourceType` The resource type of the quota.
 
 * **used**: `UnsignedInt` The current usage of the mailbox. Computation of this value is handled by the server.
 
-* **usedScope**: `Scope` The `Scope` of the current mailbox usage.
+* **limit**: `UnsignedInt` The hard limit set by this quota object. No more outgoing and ingoing messages should be
+allowed if we reach this limit. It should higher than the `warnLimit` and the `softLimit`.
+
+* **scope**: `Scope` The `Scope` of this quota.
 
-* **limit**: `UnsignedInt` The limit set by this quota object to the current mailbox.
+* **name**: `String` The name of the quota object. Useful for managing quotas and use queries for searching.
 
-* **limitScope**: `Scope` The `Scope` of the mailbox limit.
+* **datatypes**: `String[]` A list of all the data types values that are applying to this quota. This allows to assign
```

Comment:
Do we need a registry here? What if two extension defines the same type? Should we add the capability?